### PR TITLE
build publish maven plugin from CI to sonatype

### PR DESCRIPTION
**Pull request by raboof**
towards #27

### CODE REVIEW

File: maven-java/pom.xml [hunk 1]
```diff
@@ -46,34 +46,89 @@
...
+      <plugins>
+        <plugin>
+          <groupId>org.sonatype.plugins</groupId>
+          <artifactId>nexus-staging-maven-plugin</artifactId>
+          <version>1.6.7</version>
+	  <extensions>true</extensions>
+          <configuration>
+            <serverId>ossrh</serverId>
+            <nexusUrl>https://s01.oss.sonatype.org</nexusUrl>
+            <autoReleaseAfterClose>false</autoReleaseAfterClose>
```
Comment (raboof) [inline maven-java/pom.xml:66]: we could set this to 'true' eventually, but for the first release perhaps let's give ourselves a chance to check the artifacts?

File: .circleci/config.yml [hunk 2]
```diff
@@ -230,6 +230,46 @@ jobs:
...
+    steps:
+      - checkout
+      - restore_deps_cache
+      - set-sdk-version
+      - run:
+          name: Publish maven plugin and archetype
+          command: |
+            cd maven-java
+            echo $PGP_SECRET | base64 -d | gpg --import --batch
+            # It seems ~/.m2 is not actually restored from the cache?
```
Comment (octonato) [inline .circleci/config.yml:245]: We need to investigate it further. We may be doing something wrong. 

When I was bringing the maven code here, I was surprised to see that I have to call sbt publishM2 more then once in the same CI build. I was planning to call only once and reuse the local artifacts on the next jobs. Didn't work as expected.

**Review by octonato — APPROVED** [review]
LGTM

**Comment by octonato** [comment]
hmm, I could have squashed before merging. 🤦

**Pull request merged**